sdk/src/logs/logger_provider.cc

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "opentelemetry/sdk/logs/logger_provider.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace logs
+{
+LoggerProvider::LoggerProvider(std::shared_ptr<LogProcessor> processor) noexcept
+    : processor_{processor}
+{}
+
+LoggerProvider::LoggerProvider() noexcept : processor_{nullptr} {}
+
+opentelemetry::nostd::shared_ptr<opentelemetry::logs::Logger> LoggerProvider::GetLogger(
+    opentelemetry::nostd::string_view name,
+    opentelemetry::nostd::string_view options) noexcept
+{
+  // Search if a logger with the given name already exists.
```

Comment:
since lookup is probably a more frequent use case than scanning all loggers, and uniqueness by name is important, you should probably use std::unordered_set or maybe map instead of vector for storing the loggers. It's possible if you only ever have very few elements in the vector(like 3 or 4) that. you might win that way, but the performance will be bad for someone with lots. It would also save you having to manage the uniqueness constraint yourself.